Source/CarthageKit/Project.swift

```diff
@@ -926,6 +913,24 @@ public final class Project {
 					let subdirectoryPath = (CarthageProjectCheckoutsPath as NSString).appendingPathComponent(name)
 					let linkDestinationPath = relativeLinkDestinationForDependencyProject(dependency.project, subdirectory: subdirectoryPath)
 
+					let dependencyCheckoutURLResource = try? dependencyCheckoutURL.resourceValues(forKeys: [
+						.isSymbolicLinkKey,
+						.isDirectoryKey
+					])
+
+					if dependencyCheckoutURLResource?.isSymbolicLink == .some(true) {
+						_ = dependencyCheckoutURL.carthage_path.withCString(Darwin.unlink)
```

Comment:
Less comfortable with `NSFileManager.removeItem(at:)` re-reading the docs, actually: “If the URL specifies a directory, the contents of that directory are recursively removed.”

Another thought: the slim possibility of a race condition between the check, swapping in a directory, and a recursive directory deletion.

Thinking I have a stronger opinion towards `Darwin.unlink` than earlier today.